tests/test_package.py

```diff
@@ -97,6 +109,74 @@ def test_tuby_package_with_invalid_url(mock_get):
         assert "Failed to fetch: https://rubygems.org/api/v1/gems/file.json" == e_info
 
 
+@mock.patch("fetchcode.package_util.utils.get_response")
+@mock.patch("fetchcode.package_util.utils.get_github_rest")
+@mock.patch("fetchcode.package.get_cocoapod_tags")
+@mock.patch("fetchcode.package.get_hashed_path")
+def test_get_cocoapods_data_from_purl(
+    mock_get_hashed_path,
+    mock_get_cocoapod_tags,
+    mock_get_github_rest,
+    mock_get_response,
+):
+    mock_get_hashed_path.return_value = "5/5/b"
+    mock_get_cocoapod_tags.return_value = [
+        '0.1.5',
+        '0.1.4',
+        '0.1.3',
+        '0.1.2',
+        '0.1.1',
+        '0.1.0',
+    ]
+    mock_get_github_rest.return_value = load_json("tests/data/cocoapods/mock_get_github_rest_return_value.json")
+    mock_get_response.side_effect = file_json("tests/data/cocoapods/mock_get_response_side_effect.json")
+    expected_result_to_dict = file_json("tests/data/cocoapods/expected_result_to_dict.json")
+    purl = "pkg:cocoapods/ASNetworking"
+    actual_result = get_cocoapods_data_from_purl(purl)
+
+    for pkg, expected_pkg_to_dict in zip(list(actual_result), expected_result_to_dict):
+        pkg_to_json = json.dumps(pkg.to_dict())
+        expected_pkg_to_dict_json_dumps = json.dumps(expected_pkg_to_dict)
+        assert pkg_to_json == expected_pkg_to_dict_json_dumps
```

Comment:
> @pombredanne @keshav-space Like the failing tests I [commented on last week](https://github.com/nexB/fetchcode/pull/119#issuecomment-2166525075), once again all tests pass locally but 1 or 2 GitHub checks fail and the error is always the `test_get_cocoapods_data_from_purl` test, and the same supposed failure:
> 
> * we expect `"primary_language": "Swift"` but GH generates `"primary_language": null`.
> 
> This is bizarre because the various input values are mocked and thus not changing from local to GitHub. Last week after I reran the failing tests over several days, they all mysteriously passed. I'm rerunning today as well but so far I keep getting the same failure. The cause remains a mystery.
> 
> OK, many minutes have passed since I started writing this comment, during which I kept rerunning and rerunning the failed checks, and now, all of them pass. Is there some way for me to avoid this going forward? Please share ideas if you have any. 🙏 ;-)

@johnmhoran This is the test which is failing, `get_cocoapods_data_from_purl` depends on `make_head_request`. It seems like you forgot to mock the `make_head_request` function. Since `make_head_request` is not mocked, it will perform live network calls, and if the network call fails for some reason, your test will also fail.

Reply:
Thank you @keshav-space .  I see that my primary function in this test calls `construct_cocoapods_package()`, which contains a head request `utils.make_head_request(api_url)` that I overlooked here and in a related test `test_construct_cocoapods_package()`.  Thanks for your eagle eye.